pkg/client/dtclient/client.go

```diff
@@ -432,25 +435,22 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 	if err != nil {
 		return DynatraceEntity{}, fmt.Errorf("unable to generate external id: %w", err)
 	}
+
 	settingsWithExternalID, err := d.listSettings(ctx, obj.SchemaId, ListSettingsOptions{
 		Filter: func(object DownloadSettingsObject) bool {
 			return object.ExternalId == legacyExternalID
 		},
 	})
 
-	if err != nil {
```

Comment:
why did you remove this error handling? Won't we continue now in case the GET /settings API call failed?
In the case of client error here, it's likely further calls will just error as well, but still at this point data is invalid, isn't it?